src/Compilers/Core/Portable/CommandLine/ErrorLogger.cs

```diff
@@ -381,6 +389,60 @@ public string Add(DiagnosticDescriptor descriptor)
                 list.Sort((x, y) => string.CompareOrdinal(x.Key, y.Key));
                 return list;
             }
+
+            /// <summary>
+            /// Compares descriptors by the values that we write to the log and nothing else.
+            /// 
+            /// We cannot use the IEquatable implementation because it includes message format 
+            /// and we don't write it out. It also ignores tags, which we do write.
+            /// </summary>
```

Comment:
They aren't different. What I mean here is that the IEquatable implementation of DiagnosticDescriptor itself does not include CustomTags in its comparison. So this comparer is different in two ways: it includes CustomTags, and excludes MessageFormat.